utils/src/main/java/software/amazon/awssdk/utils/ToStringResult.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.utils;
+
+import java.util.Arrays;
+import software.amazon.awssdk.annotations.NotThreadSafe;
+
+/**
+ * A class to standardize implementations of {@link Object#toString()} across the SDK.
+ *
+ * <pre>{@code
+ * ToStringResult.builder("Person")
+ *               .add("name", name)
+ *               .add("age", age)
+ *               .build();
+ * }</pre>
+ */
+@NotThreadSafe
+public final class ToStringResult {
```

Comment:
Any reason not to call this `ToString` or `ToStringBuilder` ?

Reply:
`ToStringBuilder` felt weird because we couldn't follow our usual `create()`/`builder()` pattern. See: our `Item` vs `ItemBuilder` discussion we had a while back.

`ToString` felt weird because `toString` is a method, and we're not creating a method - we're creating the result of a method. Eg. `ToStringResult.create("foo")` read a bit better than `ToString.create("foo")` in a `toString` method.

`ToStringResult` sounds like it's too weird, though, so let's go with `ToString`!

Reply:
👍